diff --git a/includes/Query.php b/includes/Query.php
index ec4ade0..eaa90d6 100644
--- a/includes/Query.php
+++ b/includes/Query.php
@@ -1575,18 +1575,6 @@ class Query {
		$this->addWhere( "((SELECT count(rev_aux3.rev_page) FROM {$this->tableNames['revision']} AS rev_aux3 WHERE rev_aux3.rev_page = {$this->tableNames['page']}.page_id) <= {$option})" );
	}
 
-	/**
-	 * Set SQL for 'minoredits' parameter.
-	 *
-	 * @param mixed $option
-	 */
-	private function _minoredits( $option ) {
-		if ( isset( $option ) && $option == 'exclude' ) {
-			$this->addTable( 'revision', 'revision' );
-			$this->addWhere( 'revision.rev_minor_edit = 0' );
-		}
-	}
-
	/**
	 * Set SQL for 'minrevisions' parameter.
	 *
@@ -1867,12 +1855,13 @@ class Query {
						$this->addSelect( [ 'rev.rev_timestamp' ] );
 
						if ( !$this->revisionAuxWhereAdded ) {
-							$this->addWhere(
-								[
-									"{$this->tableNames['page']}.page_id = rev.rev_page",
-									"rev.rev_timestamp = (SELECT MAX(rev_aux.rev_timestamp) FROM {$this->tableNames['revision']} AS rev_aux WHERE rev_aux.rev_page = rev.rev_page)"
-								]
-							);
+							$this->addWhere( "{$this->tableNames['page']}.page_id = rev.rev_page" );
+
+							if ( $this->parameters->getParameter( 'minoredits' ) == 'exclude' ) {
+								$this->addWhere( "rev.rev_timestamp = (SELECT MAX(rev_aux.rev_timestamp) FROM {$this->tableNames['revision']} AS rev_aux WHERE rev_aux.rev_page = rev.rev_page AND rev_aux.rev_minor_edit = 0)" );
+							} else {
+								$this->addWhere( "rev.rev_timestamp = (SELECT MAX(rev_aux.rev_timestamp) FROM {$this->tableNames['revision']} AS rev_aux WHERE rev_aux.rev_page = rev.rev_page)" );
+							}
						}
 
						$this->revisionAuxWhereAdded = true;
