diff --git a/includes/PdfHandler.php b/includes/PdfHandler.php
index 86a328d..9ca9179 100644
--- a/includes/PdfHandler.php
+++ b/includes/PdfHandler.php
@@ -226,13 +226,13 @@ class PdfHandler extends ImageHandler {
 		);
 		$cmd .= " | " . wfEscapeShellArg(
 			$wgPdfPostProcessor,
+			"jpeg:-",
 			"-depth",
 			"8",
 			"-quality",
 			$wgPdfHandlerJpegQuality,
 			"-resize",
 			(string)$width,
-			"-",
 			$dstPath
 		);
 		$cmd .= ")";
