diff --git a/includes/PdfHandler.php b/includes/PdfHandler.php
index 0ce0faf..26abf2f 100644
--- a/includes/PdfHandler.php
+++ b/includes/PdfHandler.php
@@ -224,17 +224,31 @@ class PdfHandler extends ImageHandler {
			"-q",
			$srcPath
		);
-		$cmd .= " | " . wfEscapeShellArg(
-			$wgPdfPostProcessor,
-			"-depth",
-			"8",
-			"-quality",
-			$wgPdfHandlerJpegQuality,
-			"-resize",
-			(string)$width,
-			"-",
-			$dstPath
-		);
+		if ( wfIsWindows() ) {
+			$cmd .= " | " . wfEscapeShellArg(
+				$wgPdfPostProcessor,
+				"-",
+				"-depth",
+				"8",
+				"-quality",
+				$wgPdfHandlerJpegQuality,
+				"-resize",
+				$width,
+				$dstPath
+			);
+		} else {
+			$cmd .= " | " . wfEscapeShellArg(
+				$wgPdfPostProcessor,
+				"-depth",
+				"8",
+				"-quality",
+				$wgPdfHandlerJpegQuality,
+				"-resize",
+				$width,
+				"-",
+				$dstPath
+			);
+		}
		$cmd .= ")";
 
		wfDebug( __METHOD__ . ": $cmd\n" );
