diff --git a/src/LingoParser.php b/src/LingoParser.php
index 75271ed..adaf34d 100644
--- a/src/LingoParser.php
+++ b/src/LingoParser.php
@@ -30,6 +30,7 @@ namespace Lingo;
 
 use DOMDocument;
 use DOMXPath;
+use LogicException;
 use ObjectCache;
 use Parser;
 use StubObject;
@@ -384,6 +385,16 @@ class LingoParser {
	private function shouldParse( $parser ) {
		global $wgexLingoUseNamespaces;
 
+		try {
+			$output = $parser->getOutput();
+			if ( $output === null || $output->getText() === '' ) {
+				return false;
+			}
+		} catch( LogicException $e ) {
+			// This ParserOutput contains no text!
+			return false;
+		}
+
		if ( !( $parser instanceof Parser || $parser instanceof StubObject ) ) {
			return false;
		}
