--- includes/user/UserGroupManager.php	2023-05-26 12:50:35.231227120 +0200
+++ includes/user/UserGroupManager.fix.php	2023-05-26 12:50:17.986000000 +0200
@@ -307,7 +307,7 @@
 				$this->clearUserCacheForKind( $user, self::CACHE_EFFECTIVE );
 			}
 		}
-		return $this->userGroupCache[$userKey][self::CACHE_IMPLICIT];
+		return $this->userGroupCache[$userKey][self::CACHE_IMPLICIT] ?? [];
 	}
 
 	/**
@@ -349,7 +349,7 @@
 			$effectiveGroups = array_values( array_unique( $groups ) );
 			$this->setCache( $userKey, self::CACHE_EFFECTIVE, $effectiveGroups, $queryFlags );
 		}
-		return $this->userGroupCache[$userKey][self::CACHE_EFFECTIVE];
+		return $this->userGroupCache[$userKey][self::CACHE_EFFECTIVE] ?? [];
 	}
 
 	/**
@@ -394,7 +394,7 @@
 		}
 		$this->setCache( $userKey, self::CACHE_FORMER, $formerGroups, $queryFlags );
 
-		return $this->userGroupCache[$userKey][self::CACHE_FORMER];
+		return $this->userGroupCache[$userKey][self::CACHE_FORMER] ?? [];
 	}
 
 	/**
